src/NServiceBus.NHibernate.AcceptanceTests/Issue_164.cs

```diff
@@ -0,0 +1,187 @@
+﻿namespace NServiceBus.NHibernate.AcceptanceTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using NServiceBus.AcceptanceTesting;
+    using NServiceBus.AcceptanceTests;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NServiceBus.Config;
+    using NServiceBus.Config.ConfigurationSource;
+    using NServiceBus.Configuration.AdvanceExtensibility;
+    using NServiceBus.Features;
+    using NServiceBus.Persistence;
+    using NServiceBus.Timeout.Core;
+    using NUnit.Framework;
+
+    public class Issue_164 : NServiceBusAcceptanceTest
+    {
+        private static string CommandTagHeader = "TestTag.HeaderName";
+
+        [Test]
+        public void Timeout_does_not_get_stuck_when_persister_commit_takes_longer_than_dispatch_of_another_later_timeout()
+        {
+            //TODO: trick TM to query timeouts for frequently. Tweak GetNextChuns in adapter
+            //TODO: make clean-up interval and clean-up period configurable and override it in this test
+            var ctx = new Context();
+
+            Scenario.Define(ctx)
+                .WithEndpoint<Endpoint>(b =>
+                    b.When(bus =>
+                    {
+                        ctx.CommandATag = "MessageA: " + Guid.NewGuid().ToString();
+                        ctx.CommandBTag = "MessageB: " + Guid.NewGuid().ToString();
```

Comment:
The tag is not enough. It is not easy to purge TM satellite queues and persister on startup. I'll make it guid only.